### PR TITLE
minikube script for travis

### PR DESCRIPTION
**Pull request by AdheipSingh**

### CODE REVIEW

File: scripts/travis/minikube.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+ # Download minikube.
+- curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+- mkdir -p $HOME/.kube $HOME/.minikube
+- touch $KUBECONFIG
+- sudo minikube start --profile=minikube --vm-driver=none --kubernetes-version=v1.18.1
+- minikube update-context --profile=minikube
+- "sudo chown -R travis: /home/travis/.minikube/"
+- eval "$(minikube docker-env --profile=minikube)" && export DOCKER_CLI='docker'
+- JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until kubectl -n kube-system get pods -lk8s-app=kube-dns -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1;echo "waiting for kube-dns to be available"; kubectl get pods --all-namespaces; done
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: lets reduce the length of the line, we can make it multiline

File: scripts/travis/minikube.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+echo " minikube setup"
+echo "==============================================================================================================="
+
+ # Download minikube.
+- curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+- mkdir -p $HOME/.kube $HOME/.minikube
+- touch $KUBECONFIG
+- sudo minikube start --profile=minikube --vm-driver=none --kubernetes-version=v1.18.1
+- minikube update-context --profile=minikube
+- "sudo chown -R travis: /home/travis/.minikube/"
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: we can probably add some echo for diff steps

File: scripts/travis/minikube.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+echo "==============================================================================================================="
+echo " minikube setup"
+echo "==============================================================================================================="
+
+ # Download minikube.
+- curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]:
```suggestion
curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 \
     && chmod +x minikube \
     && sudo mv minikube /usr/local/bin/
```

File: scripts/travis/minikube.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+- mkdir -p $HOME/.kube $HOME/.minikube
+- touch $KUBECONFIG
+- sudo minikube start --profile=minikube --vm-driver=none --kubernetes-version=v1.18.1
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: Make version as variable, so that its easy to upgrade.

**Review by himadrisingh — CHANGES_REQUESTED** [review]

File: scripts/travis/minikube.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+echo "==============================================================================================================="
+echo " minikube setup"
+echo "==============================================================================================================="
+
+curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+mkdir -p $HOME/.kube $HOME/.minikube
+touch $KUBECONFIG
+sudo minikube start --profile=minikube --vm-driver=none --kubernetes-version=v1.18.1
+minikube update-context --profile=minikube
+"sudo chown -R travis: /home/travis/.minikube/"
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: why is this in quotes?

File: scripts/travis/minikube.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+echo "==============================================================================================================="
+echo " minikube setup"
+echo "==============================================================================================================="
+
+curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: length

File: scripts/travis/minikube.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+mkdir -p $HOME/.kube $HOME/.minikube
+touch $KUBECONFIG
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh:20]: where is `KUBECONFIG` defined? and whats the default value.

File: scripts/travis/minikube.sh (hunk 5 above)
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: what are we doing here

File: scripts/travis/minikube.sh [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+sudo minikube start --profile=minikube --vm-driver=none --kubernetes-version=v1.18.1
+minikube update-context --profile=minikube
+"sudo chown -R travis: /home/travis/.minikube/"
+eval "$(minikube docker-env --profile=minikube)" && export DOCKER_CLI='docker'
+JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; \
+  until kubectl -n kube-system get pods -lk8s-app=kube-dns -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; \
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: this can be a separate line

File: scripts/travis/minikube.sh [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  do sleep 1; \
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: lets sleep for 5s or something

File: scripts/travis/minikube.sh [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  echo "waiting for kube-dns to be available"; \ 
+  kubectl get pods --all-namespaces; \
+  done;
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]: add some newlines between diff operations

**Review by himadrisingh — CHANGES_REQUESTED** [review]

File: scripts/travis/minikube.sh (hunk 7 above)
Comment (AdheipSingh) [reply scripts/travis/minikube.sh:20]:
```
 CHANGE_MINIKUBE_NONE_USER=true
 MINIKUBE_WANTUPDATENOTIFICATION=false
 MINIKUBE_WANTREPORTERRORPROMPT=false
 MINIKUBE_HOME=$HOME
 CHANGE_MINIKUBE_NONE_USER=true
 KUBECONFIG=$HOME/.kube/config
```
These are global vars declared in travis

File: scripts/travis/minikube.sh (hunk 10 above)
Comment (AdheipSingh) [reply scripts/travis/minikube.sh]: removing this as it would print a lot on each iteration

File: scripts/travis/minikube.sh (hunk 7 above)
Comment (himadrisingh) [reply scripts/travis/minikube.sh:20]: hmm do we need those in travis settings, these will be common to all right. we can put in here in the script.

File: scripts/travis/minikube.sh [hunk 11]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -e
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh:2]:
```suggestion
set -e

```

File: scripts/travis/minikube.sh [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+KUBE_VERSION=1.18.1
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh:4]:
```suggestion
KUBE_VERSION=1.18.1

```

File: scripts/travis/minikube.sh [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+echo "==============================================================================================================="
+echo " minikube setup"
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]:
```suggestion
echo " Setup minikube"
```

File: scripts/travis/minikube.sh [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+echo "==============================================================================================================="
+curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]:
```suggestion
curl -Lo /usr/local/bin/minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && \
    chmod +x /usr/local/bin/minikub
```

File: scripts/travis/minikube.sh [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+chmod +x minikube
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]:
```suggestion
```

File: scripts/travis/minikube.sh [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+sudo mv minikube /usr/local/bin/
```
Comment (himadrisingh) [inline scripts/travis/minikube.sh]:
```suggestion

```

**Pull request merged**